From 852d6d52b438b8015761ebf10f12eb583160ffa5 Mon Sep 17 00:00:00 2001
From: ywan171 <yi.a.wang@intel.com>
Date: Thu, 26 Jul 2012 08:00:06 +0800
Subject: [PORT FROM MAIN]NuPlayerRender: fix a/v sync issue when playback only video track clip

BZ: 48710

orig patch:
43398:fix google bugs
If only video track exists, mAnchorTimeMediaUs retrieve system time asa/v sync time base.
If playing is paused, we should clear mAnchorTimeMediaUsand reset it when resume playing

Change-Id: If71b0e83b0bbf0d9f7687675e5b0375e7eff8595
Orig-Change-Id: I9490177465656d28c54143581bbdc0e8bc2e32e3
Signed-off-by: ywan171 <yi.a.wang@intel.com>
---
 .../nuplayer/NuPlayerRenderer.cpp                  |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp b/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp
index 404b56f..acd2658 100644
--- a/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp
+++ b/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp
@@ -660,6 +660,9 @@ void NuPlayer::Renderer::onPause() {
 
     if (mHasAudio) {
         mAudioSink->pause();
+    } else {
+        // if no audio, we need to clear mAnchorTimeMediaUs, reset it when first video buffer comes
+        mAnchorTimeMediaUs = -1;
     }
 
     ALOGV("now paused audio queue has %d entries, video has %d entries",
-- 
1.7.4.1

